sdk/src/lib.rs

```diff
@@ -116,7 +143,7 @@ where
     F: Message + Default,
     D: Message + Default,
 {
-    type Item = Result<StreamDataResponse, Box<dyn std::error::Error>>;
+    type Item = Result<Vec<D>, Box<dyn std::error::Error>>;
```

Comment:
Let's use the data message defined earlier.

```suggestion
    type Item = Result<DataMessage<D>, Box<dyn std::error::Error>>;
```